### PR TITLE
fix: Use meta operator API for precheck AIO discovery

### PR DESCRIPTION
**Pull request by c-ryan-k**
---
This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.

Thank you for contributing to Azure IoT Operations tooling!

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

Intent for Production

- [ ] It is expected that pull requests made to default or core branches such as `dev` or `main` are of production grade. Corollary to this, any merged contributions to these branches may be deployed in a public release at any given time. By checking this box, you agree and commit to the expected production quality of code.

Basic expectations

- [ ] If introducing new functionality or modified behavior, are they backed by unit and/or integration tests?
- [ ] In the same context as above are command names and their parameter definitions accurate? Do help docs have sufficient content?
- [ ] Have **all** the relevant unit **and** integration tests pass? i.e. `pytest <project root> -vv`. Please provide evidence in the form of a screenshot showing a succesful run of tests locally OR a link to a test pipeline that has been run against the change-set.
- [ ] Have linter checks passed using the `.pylintrc` and `.flake8` rules? Look at the CI scripts for example usage.
- [ ] Have extraneous print or debug statements, commented out code-blocks or code-statements (if any) been removed from the surface area of changes?
- [ ] Have you made an entry in HISTORY.rst which concisely explains your user-facing feature or change?

Azure IoT Operations CLI maintainers reserve the right to enforce any of the outlined expectations.

A PR is considered **ready for review** when all basic expectations have been met (or do not apply).

### CODE REVIEW

File: azext_edge/edge/commands_edge.py [hunk 1]
```diff
@@ -61,9 +61,11 @@ def check(
     load_config_context(context_name=context_name)
     from .providers.checks import run_checks
 
-    # by default - run prechecks if AIO is not deployed
-    run_pre = not ORC_API_V1.is_deployed() if pre_deployment_checks is None else pre_deployment_checks
-    run_post = True if post_deployment_checks is None else post_deployment_checks
+    aio_deployed = MQ_ACTIVE_API.is_deployed()
```
Comment (vilit1) [inline azext_edge/edge/commands_edge.py:64]: can we use meta? or would mq still be fine (since it will be there)

**Review by vilit1 — APPROVED** [review]
tbh there are 3 states now:
1. before init
2. after init but before create
3. after create
as long as we have the divide of pre vs post between 2/3 (so 1 and 2 are pre but 3 is post) I am good

File: azext_edge/edge/commands_edge.py (hunk 1 above)
Comment (c-ryan-k) [reply azext_edge/edge/commands_edge.py:64]: I just needed something non-orc, i'm sure meta would be fine too but this just unblocks us for now

**Comment by c-ryan-k** [comment]
> tbh there are 3 states now:
> 
> 1. before init
> 2. after init but before create
> 3. after create
>    as long as we have the divide of pre vs post between 2/3 (so 1 and 2 are pre but 3 is post) I am good

From the check perspective there is no `2` - you either have AIO deployed or not. We don't have any checks for foundation components (init only) so it doesn't really make sense to account for that state yet.

File: azext_edge/edge/commands_edge.py (hunk 1 above)
Comment (Elsie4ever) [reply azext_edge/edge/commands_edge.py:64]: lol got the same place edited using meta because I'm removing orc😆

Comment (digimaun) [reply azext_edge/edge/commands_edge.py:64]: meta is probably the most appropriate

File: azext_edge/tests/edge/checks/int/test_pre_post_int.py [hunk 2]
```diff
@@ -39,8 +39,8 @@ def test_check_pre_post(init_setup, post, pre):
 
     if pre is None and not post:
         try:
-            aio_check = run("kubectl api-resources --api-group=orchestrator.iotoperations.azure.com")
-            expected_pre = "orchestrator.iotoperations.azure.com" not in aio_check
+            aio_check = run("kubectl api-resources --api-group=iotoperations.azure.com")
+            expected_pre = "iotoperations.azure.com" not in aio_check
         except CLIInternalError:
             from azext_edge.edge.providers.edge_api.orc import ORC_API_V1
```
Comment (vilit1) [inline azext_edge/tests/edge/checks/int/test_pre_post_int.py:45]: ORC strikes again

**Review by Elsie4ever — APPROVED** [review]

**Review by vilit1 — APPROVED** [review]

**Pull request merged**